src/SSHLibrary/pythonclient.py

```diff
@@ -80,6 +81,26 @@ def _login_with_public_key(self, username, key_file, password):
                                 allow_agent=False, timeout=float(self.config.timeout))
         except paramiko.AuthenticationException:
             raise SSHClientException
+        self.banner = self.client.get_transport().get_banner()
+
+    def get_banner(self):
+        return self.banner
+
+    @staticmethod
+    def get_pre_login_banner(host, port=None):
+        if not port:
+            port = 22
+        if host:
```

Comment:
1. Names `get_banner()` and `get_pre_login_banner()` don't make it that clear why we have two separate methods for getting the banner. Perhaps it would be a bit more clear if the latter was named to something like `get_banner_without_login`.

2. Why not just have `port=22` in the signature?

3. No need to handle `host` being `None` here anymore. It's should be checked already on the library side. Also, if it was to be checked here, should use 

    ```python
    if not host:
        raise SSHClientException("Missing value for argument 'host'!")
    # Rest of the code without `else` block
    ```
    to avoid having a long if/else  block.